test/extended/authorization/podsecurity_admission.go

```diff
@@ -189,14 +193,302 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity][Feature:SCC]", func() {
+			{"baseline-labelsync-off", "false", "", "", "baseline", "v1.25", "baseline", "v1.25", false},
+			{"privileged-labelsync-off", "false", "", "", "privileged", "v1.25", "privileged", "v1.25", false},
+
+			{"restricted-labelsync-on", "true", "restricted", "v1.24", "restricted", "v1.25", "restricted", "v1.24", true},
+			{"baseline-labelsync-on", "true", "restricted", "v1.24", "baseline", "v1.25", "restricted", "v1.24", true},
+			{"privileged-labelsync-on", "true", "restricted", "v1.24", "privileged", "v1.25", "restricted", "v1.24", true},
+
+			{"restricted-labelsync-undef", "", "restricted", "v1.24", "restricted", "v1.25", "restricted", "v1.25", true}, // sccChangesPSa = true because the field hasn't been taken over as the update did not change the label
+			{"baseline-labelsync-undef", "", "restricted", "v1.24", "baseline", "v1.25", "baseline", "v1.25", false},
+			{"privileged-labelsync-undef", "", "restricted", "v1.24", "privileged", "v1.25", "privileged", "v1.25", false},
+
+			// openshift- prefixed namespaces; special case when podSecurityLabelSync label is undefined
+			{"openshift-e2e-restricted-labelsync-off", "false", "", "", "restricted", "v1.25", "restricted", "v1.25", false},
+			{"openshift-e2e-baseline-labelsync-off", "false", "", "", "baseline", "v1.25", "baseline", "v1.25", false},
+			{"openshift-e2e-privileged-labelsync-off", "false", "", "", "privileged", "v1.25", "privileged", "v1.25", false},
+
+			{"openshift-e2e-restricted-labelsync-on", "true", "restricted", "v1.24", "restricted", "v1.25", "restricted", "v1.24", true},
+			{"openshift-e2e-baseline-labelsync-on", "true", "restricted", "v1.24", "baseline", "v1.25", "restricted", "v1.24", true},
+			{"openshift-e2e-privileged-labelsync-on", "true", "restricted", "v1.24", "privileged", "v1.25", "restricted", "v1.24", true},
+
+			{"openshift-e2e-restricted-labelsync-undef", "", "", "", "restricted", "v1.25", "restricted", "v1.25", false},
+			{"openshift-e2e-baseline-labelsync-undef", "", "", "", "baseline", "v1.25", "baseline", "v1.25", false},
+			{"openshift-e2e-privileged-labelsync-undef", "", "", "", "privileged", "v1.25", "privileged", "v1.25", false},
+		} {
+			g.By(testCase.name, func() {
+				ns := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   testCase.name,
+						Labels: map[string]string{},
+					},
+				}
+
+				if testCase.podSecurityLabelSync != "" {
+					ns.ObjectMeta.Labels["security.openshift.io/scc.podSecurityLabelSync"] = testCase.podSecurityLabelSync
+				}
+
+				// create the test namespace without any PSa labels; podSecurityLabelSync label according to each test case
+				ns, err := createOrUpdateNamespace(ctx, oc, ns, metav1.CreateOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// assert expectations after creation depending on the podSecurityLabelSync label
+				o.Expect(ns.GetLabels()[psapi.EnforceLevelLabel]).To(o.Equal(testCase.expectedPSaLevelAfterCreate))
```

Comment:
you can access labels directly, the metadata's `Get*` methods are mostly there for passing the meta objects as interfaces

Reply:
fixed in 5efd44f787c4220c7cbf3b00a16e58b8526aec10